src/coreclr/src/jit/gentree.cpp

```diff
@@ -15563,6 +15563,45 @@ GenTree* Compiler::gtBuildCommaList(GenTree* list, GenTree* expr)
     }
 }
 
+//------------------------------------------------------------------------
+// gtExtractSideEffList: Fixup a side-effect tree that produces a (dead) struct result
+//
+// Arguments:
+//    expr       - the side effect expression tree
```

Comment:
Add comments for `block` and `clsHnd`?

Was puzzled why block was needed, but see it's apparently just to mark the block as containing a null check, which seems fine.